Automattic/wp-calypso
# Products: Remove formatProduct from G Suite functions

#### Changes proposed in this Pull Request

In the vein of https://github.com/Automattic/wp-calypso/pull/58320, https://github.com/Automattic/wp-calypso/pull/58270, https://github.com/Automattic/wp-calypso/pull/58197, https://github.com/Automattic/wp-calypso/pull/58136, #58080, and #58086, this PR removes more uses of the `formatProduct` function in the `calypso-products` package because it makes it difficult to determine types and in some cases is unnecessary.

In this PR specifically, we change the following functions and convert them to TypeScript. Except for `isGoogleWorkspaceExtraLicence`, none of these changes modify the functionality of these functions, except that they now always require an object (where they previously would accept anything).

- `isGSuite` (unused)
- `isGSuiteOrExtraLicense` (unused)
- `isGoogleWorkspace`
- `isGSuiteOrExtraLicenseOrGoogleWorkspace`
- `isGSuiteOrGoogleWorkspace`
- `isGoogleWorkspaceExtraLicence`

In addition, this changes `isGoogleWorkspaceExtraLicence` to support both products and purchases, and then uses it to replace the function of the same name in `client`. This is not totally necessary, but it seems to me that it will reduce confusion about these otherwise identical functions.

#### Testing instructions

For most of these functions, no specific testing should be necessary, other than making sure that all the call-sites pass in an object.

For `isGoogleWorkspaceExtraLicence`, you could look at the testing instructions for https://github.com/Automattic/wp-calypso/pull/50670 but just checking that all call sites are importing the function from the package should be sufficient.
